test/rule_test_support.dart

```diff
@@ -49,12 +61,164 @@ class AnalysisOptionsFileConfig {
   }
 }
 
+/// A description of a diagnostic that is expected to be reported.
+class ExpectedDiagnostic {
+  /// An empty array of diagnostic descriptors used when no errors are expected.
+  static List<ExpectedDiagnostic> noDiagnostics = <ExpectedDiagnostic>[];
```

Comment:
I don't know whether this was in the original source, but there's no value in Dart to defining constants like this. It's better to add `const` before the right-hand side and then just inline the expression everywhere.

Reply:
It was in the original source.  My guess is the intention was to make it clear that an empty list was meant to be unpopulated and not an accident but I don't feel strongly so I'll remove.